consensus/src/simplex/types.rs

```diff
@@ -71,9 +71,12 @@ impl<T: Attributable> AttributableMap<T> {
     /// if it has not been added yet.
     ///
     /// Returns `true` if the item was inserted, `false` if an item from this
-    /// signer already exists.
+    /// signer already exists or if the signer index is out of bounds.
     pub fn insert(&mut self, item: T) -> bool {
         let index = item.signer() as usize;
+        if index >= self.data.len() {
```

Comment:
I kind of think we should panic here? If this happens, you've misconfigured AttributableMap?

Reply:
Ah, if we're validating messages against this value that may be a different story... (would also need to add to `get`)?

Reply:
This should never happen with our current usage (in batcher), we always start by checking the signer public key against the participants set to get a valid index, so this will never be out-of-bounds. Still, it probably makes sense to expose a safer API like this. `get()` is already handling out-of-bounds.